instrumentation/src/main/java/io/opentelemetry/android/instrumentation/crash/CrashReporter.java

```diff
@@ -38,16 +45,47 @@ public void installOn(InstrumentedApplication instrumentedApplication) {
                 Thread.getDefaultUncaughtExceptionHandler();
         Thread.setDefaultUncaughtExceptionHandler(
                 new CrashReportingExceptionHandler(
-                        buildInstrumenter(instrumentedApplication.getOpenTelemetrySdk()),
-                        instrumentedApplication.getOpenTelemetrySdk().getSdkTracerProvider(),
+                        buildInstrumenter(
+                                instrumentedApplication
+                                        .getOpenTelemetrySdk()
+                                        .getSdkLoggerProvider()),
+                        instrumentedApplication.getOpenTelemetrySdk().getSdkLoggerProvider(),
                         existingHandler));
     }
 
-    private Instrumenter<CrashDetails, Void> buildInstrumenter(OpenTelemetry openTelemetry) {
-        return Instrumenter.<CrashDetails, Void>builder(
-                        openTelemetry, "io.opentelemetry.crash", CrashDetails::spanName)
-                .addAttributesExtractor(new CrashDetailsAttributesExtractor())
-                .addAttributesExtractors(additionalExtractors)
-                .buildInstrumenter();
+    private void emitCrashEvent(Logger crashReporter, CrashDetails crashDetails) {
+        Throwable throwable = crashDetails.getCause();
+        Thread thread = crashDetails.getThread();
+        AttributesBuilder attributesBuilder =
+                Attributes.builder()
+                        .put(SemanticAttributes.EXCEPTION_ESCAPED, true)
+                        .put(SemanticAttributes.THREAD_ID, thread.getId())
+                        .put(SemanticAttributes.THREAD_NAME, thread.getName())
+                        .put(SemanticAttributes.EXCEPTION_MESSAGE, throwable.getMessage())
+                        .put(SemanticAttributes.EXCEPTION_STACKTRACE, stackTraceToString(throwable))
+                        .put(SemanticAttributes.EXCEPTION_TYPE, throwable.getClass().getName());
+
+        for (AttributesExtractor<CrashDetails, Void> extractor : additionalExtractors) {
+            extractor.onStart(attributesBuilder, Context.current(), crashDetails);
+        }
+
+        crashReporter.logRecordBuilder().setAllAttributes(attributesBuilder.build()).emit();
+    }
+
+    private String stackTraceToString(Throwable throwable) {
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+
+        throwable.printStackTrace(pw);
+        pw.flush();
+
+        return sw.toString();
```

Comment:
for some reason Timber allocates only 256 and flush manually, see https://github.com/JakeWharton/timber/blob/dfc4e70bd1a4b4e1e136647b96fc9c9dbcb41f65/timber/src/main/java/timber/log/Timber.kt#L175-L181
not sure there's any performance gain.

Reply:
That's quite interesting, the 256 initial buffer size seems related to performance indeed. I'm usually not keen on doing premature optimizations, though I think, based on how popular Timber is, I guess it's worth following their approach.

Regarding the `and flush manually` part I'm a bit confused because it seems like that's the default behavior of `PrintWriter`, so I'm not sure why they had to use that constructor if they didn't need to change the defaults.

Reply:
I've added the 256 initial size.

Reply:
probably because you call `flush` manually anyway? so makes sense that the autoFlush=false, unless the default is false already

Reply:
Yeah, the default is false, that's why I'm a little confused about the code in Timber 😅